pkg/apiserver/api/v1/instance.proto

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package kubecarrier.api.v1;
+option go_package = "v1";
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+
+import "meta.proto";
+
+message Instance {
+  ObjectMeta metadata = 1;
+  string spec = 2;
+  string status = 3;
+}
+
+message InstanceList {
+  ListMeta metadata = 1;
+  repeated Instance items = 2;
+}
+
+message InstanceGetRequest {
+  // Offering name, i.e. couchdb.eu-west-1.team-a
+  string offering = 1;
+  // Version of the resource
+  string version = 2;
+  // Name of the offering instance
+  string name = 3;
+  // Account indicate namespace of the project/account
+  string account = 4;
+}
+
+message InstanceDeleteRequest {
+  // Offering name, i.e. couchdb.eu-west-1.team-a
+  string offering = 1;
+  // Version of the resource
+  string version = 2;
+  // Name of the offering instance
+  string name = 3;
+  // Account indicate namespace of the project/account
+  string account = 4;
+}
+
+message InstanceListRequest {
+  // Offering name, i.e. couchdb.eu-west-1.team-a
+  string offering = 1;
```

Comment:
Can I list all instances within an account? Or is this something you'd like handling on the front-end?

Reply:
Do you mean all instances of all different offerings? In that case, no, current API can't do that, we can add this later, but it will be separate endpoint(`/v1/accounts/{account}/instances/`

Reply:
ok